Make form read-only on "done" actions

paulfalgout opened the pull request:
Shortcut Story ID: [sc-34969]

Not doing much here.  This is a policy and not a permission, so it just prompts the form how to render based on the status of the action at render time, but it is not enforced beyond that.

paulfalgout committed:
Make default action stub state in progress

This is because we have behavioral changes on the done states in the near future

paulfalgout commented on test/support/api/actions.js:
```diff
@@ -57,6 +57,7 @@ Cypress.Commands.add('routeAction', (mutator = _.identity) => {
     response() {
       const apiData = generateData.call(this);
       apiData.data = _.sample(apiData.data);
+      apiData.data.relationships.state.data.id = '33333';
```
This is by far the bigger change on this PR. I don't _think_ it is a problem, though we'll se a lot less to-do actions randomly around.

nmajor25 commented on test/integration/forms/action.js:
```diff
@@ -1030,21 +1030,58 @@ context('Patient Action Form', function() {
+      .routePatient()
+      .routeFormByAction()
+      .routeFormDefinition()
+      .routeActionActivity()
```
Do we need `routePatient()` or `routeActionActivity()` here?

paulfalgout replied:
ah right this should be a `routesFor` this branch is actually really old.. will update

paulfalgout committed:
Make forms on “done” actions read only

paulfalgout committed:
Make read-only status forms render readOnly in form.io

Previously we were only disabling the app submit/update button, but form definitions may now include submit buttons, so the forms themselves need to be rendered readOnly.
